src/System.Private.ServiceModel/tests/Scenarios/Client/ExpectedExceptions/ExpectedExceptionTests.cs

```diff
@@ -248,6 +247,144 @@ public static void FaultException_Throws_WithFaultDetail()
+                    error += String.Format("\r\nInnerException:\r\n{0}", e.InnerException.ToString());
+                errorBuilder.AppendLine(error);
+            }
+            else
+            {
+                FaultException<FaultDetail> faultException = (FaultException<FaultDetail>)(e);
+                string actualFaultMsg = ((FaultDetail)(faultException.Detail)).Message;
+                if (actualFaultMsg != faultMsg)
+                {
+                    errorBuilder.AppendLine(string.Format("Expected Fault Message: {0}, actual: {1}", faultMsg, actualFaultMsg));
+                }
+            }
+
+            Assert.True(errorBuilder.Length == 0, string.Format("Test Scenario: FaultException_Throws_WithFaultDetail FAILED with the following errors: {0}", errorBuilder));
+            return;
+        }
+
+        Assert.True(false, "Expected FaultException<FaultDetail> exception, but no exception thrown.");
+    }
+
+    [Theory]
+    [OuterLoop]
+    public static void FaultException_Throws_FaultDetail2()
+    {
+        string faultMsg = "Test Fault Exception";
+        StringBuilder errorBuilder = new StringBuilder();
+
+        try
+        {
+            BasicHttpBinding binding = new BasicHttpBinding();
+            using (ChannelFactory<IWcfService> factory = new ChannelFactory<IWcfService>(binding, new EndpointAddress(Endpoints.HttpBaseAddress_Basic)))
+            {
+                IWcfService serviceProxy = factory.CreateChannel();
+                serviceProxy.TestFaults(faultMsg, false);
+            }
+        }
+        catch (Exception e)
+        {
+            if (e.GetType() != typeof(FaultException<FaultDetail2>))
+            {
+                string error = string.Format("Expected exception: {0}, actual: {1}\r\n{2}",
+                                             "FaultException<FaultDetail2>", e.GetType(), e.ToString());
+                if (e.InnerException != null)
+                    error += String.Format("\r\nInnerException:\r\n{0}", e.InnerException.ToString());
+                errorBuilder.AppendLine(error);
+            }
+            else
+            {
+                FaultException<FaultDetail2> faultException = (FaultException<FaultDetail2>)(e);
+                string actualFaultMsg = ((FaultDetail2)(faultException.Detail)).Message;
+                if (actualFaultMsg != faultMsg)
+                {
+                    errorBuilder.AppendLine(string.Format("Expected Fault Message: {0}, actual: {1}", faultMsg, actualFaultMsg));
+                }
+            }
+
+            Assert.True(errorBuilder.Length == 0, string.Format("Test Scenario: FaultException_Throws_WithFaultDetail2 FAILED with the following errors: {0}", errorBuilder));
+            return;
+        }
+
+        Assert.True(false, "Expected FaultException<FaultDetail2> exception, but no exception thrown.");
+    }
+
+    [Theory]
+    [OuterLoop]
+    public static void FaultException_Echo_Or_Throws_Fault_Exception()
```

Comment:
I've moved these tests to Contract\Fault\FaultExceptionTests.cs.
This test was intended to test that round trip of known type works when there is both [ServiceKnownType] and [FaultContract].